test/benchmark/Benchmark.t.sol

```diff
@@ -0,0 +1,123 @@
+
+    uint128 internal immutable AMOUNT_PER_SEGMENT = 100e18;
+    uint128 internal immutable AMOUNT_PER_TRANCHE = 100e18;
+    uint256 internal immutable STREAM_1 = 50;
+    uint256 internal immutable STREAM_2 = 51;
+    uint256 internal immutable STREAM_3 = 52;
+    uint256 internal immutable STREAM_4 = 53;
+    uint256 internal immutable STREAM_5 = 54;
+
+    /// @dev The directory where the benchmark files are stored.
+    string internal benchmarksDir = "benchmarks/";
+
+    /// @dev The path to the file where the benchmarks are stored.
+    string internal benchmarksFile;
+
+    string internal contentToAppend;
+
+    ISablierV2Lockup internal lockup;
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                  SET-UP FUNCTION
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function setUp() public virtual override {
+        super.setUp();
+
+        deal({ token: address(dai), to: users.sender, give: type(uint256).max });
+        resetPrank({ msgSender: users.sender });
+
+        // Create the first streams in each Lockup contract to initialize all the variables.
+        _createFewStreams();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                    GAS BENCHMARKS FOR COMMON IMPLEMENTATIONS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function gasBurn() internal {
+        lockup.withdraw(STREAM_1, users.recipient, defaults.DEPOSIT_AMOUNT());
+
+        uint256 beforeGas = gasleft();
+        lockup.burn(STREAM_1);
+        uint256 afterGas = gasleft();
+
+        contentToAppend = string.concat("| `burn` | ", vm.toString(beforeGas - afterGas), " |");
+
+        // Append the content to the file.
+        _appendToFile(benchmarksFile, contentToAppend);
+    }
+
+    function gasCancel() internal {
+        uint256 beforeGas = gasleft();
+        lockup.cancel(STREAM_2);
+        uint256 afterGas = gasleft();
+
+        contentToAppend = string.concat("| `cancel` | ", vm.toString(beforeGas - afterGas), " |");
+
+        // Append the content to the file.
+        _appendToFile(benchmarksFile, contentToAppend);
+    }
+
+    function gasRenounce() internal {
+        uint256 beforeGas = gasleft();
+        lockup.renounce(STREAM_3);
+        uint256 afterGas = gasleft();
+
+        contentToAppend = string.concat("| `renounce` | ", vm.toString(beforeGas - afterGas), " |");
+
+        // Append the content to the file.
+        _appendToFile(benchmarksFile, contentToAppend);
+    }
+
+    function gasWithdraw_ByRecipient() internal {
+        uint256 beforeGas = gasleft();
+        lockup.withdraw(STREAM_4, users.alice, defaults.WITHDRAW_AMOUNT());
+        uint256 afterGas = gasleft();
+
+        contentToAppend = string.concat("| `withdraw` (by Recipient) | ", vm.toString(beforeGas - afterGas), " |");
```

Comment:
Calling it "withdraw to recipient" does not make it clear who is the caller. It could be anyone. Then we would have to handle three different cases such as
- withdraw to recipient by sender / anyone
- withdraw to recipient by recipient
- withdraw to anyone by recipient

"Withdraw by recipient" and "withdraw" confirm that the first case is where withdraw is called by the recipient and the second one is where it is not called by the recipient.

wdyt?

Reply:
Ok. Lets have it your way.